Havtorn/Source/Engine/ECS/Systems/DebugShapeSystem.cpp

```diff
@@ -89,27 +89,162 @@ namespace Havtorn
 				return;
 
 			const std::vector<Ref<SEntity>>& entities = Instance->Scene->GetEntities();
+			
 			const U64 shapeIndex = entities[entityIndex]->GetComponentIndex(EComponentType::DebugShapeComponent);
 			std::vector<Ref<SDebugShapeComponent>>& debugShapes = Instance->Scene->GetDebugShapeComponents();
 			SetSharedDataForShape(debugShapes[shapeIndex], color, lifeTimeSeconds, useLifeTime, thickness, ignoreDepth);
 			debugShapes[shapeIndex]->VertexBufferIndex = Utility::VertexBufferPrimitives::GetVertexBufferIndex<U8>(EVertexBufferPrimitives::LineShape);
-			debugShapes[shapeIndex]->VertexCount = Utility::VertexBufferPrimitives::GetVertexCount<U8>(EVertexBufferPrimitives::LineShape);
+			debugShapes[shapeIndex]->IndexCount = Utility::VertexBufferPrimitives::GetIndexCount<U8>(EVertexBufferPrimitives::LineShape);
```

Comment:
I think the usability or accessibility would increase if we found a common denominator between Utility::VertexBufferPrimitives and UGraphicsUtils or moved one to the other. 

In addition, it's annoying that some namespaces have prefixes (UGraphicsUtils) while others don't (GeometryPrimitives), we need to figure out what to do about that, I think it can depend on what you want it to look like in code

Please think about this, I don't think we need separate namespaces for these things

Reply:
Good catch. I hadn't considered using UGraphicsUtils for this as well. Agreed that it would improve usability. Will move Utility::VertexBufferPrimitives to UGraphicsUtils.